central/detection/service/service_impl.go

```diff
@@ -431,13 +441,35 @@ func (s *serviceImpl) DetectDeployTimeFromYAML(ctx context.Context, req *apiV1.D
 		if run != nil {
 			runs = append(runs, run)
 		}
+		an, err := s.getAppliedNetpolsForDeployment(ctx, eCtx, d)
+		if err != nil {
+			log.Warnf("Failed to get Network Policies for deployment %s, continuing without. Error: %v", d.GetName(), err)
+			continue
+		}
+
+		if remarks[d.Name] == nil {
+			remarks[d.Name] = &apiV1.DeployDetectionRemark{}
+		}
```

Comment:
Why do we need that? We are replacing the value stored in `remarks[d.Name]` in the next line.

Reply:
In the next line, we're only setting the `AppliedNetworkPolicies` of the `DeployDetectionRemark` struct.
If setting the permission level in line 426,ff. worked, we can add to that struct in 453. 
If setting it didn't work, e.g. Sensor enrichment failed, we would otherwise try to set `AppliedNetworkPolicies` on an uninitialized struct, wouldn't we?
That said, I've changed this line to also set the Name when creating the remark, as that would also be missing.

Reply:
That makes total sense now. I could bet that I saw something like:

```go
remarks[d.Name]= getPolicyNamesAsSlice(an.Policies)
```

in the next line